src/status_im/transport/message/core.cljs

```diff
@@ -12,10 +13,24 @@
             [status-im.utils.fx :as fx]
             [taoensso.timbre :as log]))
 
+(defn unwrap-message
+  "Extract message from new payload {:id some-id :message some-message}
+  or old (just plain message)"
+  [js-message]
+  (let [clj-message (js->clj js-message :keywordize-keys true)
+        {:keys [message id]} clj-message]
+    {:message (or message clj-message)
+     :raw-payload (if message
+                    (o/get js-message "message")
+                    js-message)
+     :id id}))
+
 (fx/defn receive-message
   [cofx now-in-s filter-chat-id js-message]
   (let [blocked-contacts (get-in cofx [:db :contacts/blocked] #{})
-        {:keys [payload sig timestamp ttl]} (js->clj js-message :keywordize-keys true)
+        {{:keys [payload sig timestamp ttl]} :message
+         dedup-id :id
```

Comment:
Could you add a docstring to explain in the code what the dedup-id is for and where it is coming from, not sure where is the best place to do that maybe here.

Reply:
sure, good idea